src/lib/xmlparse.rs

```diff
@@ -1912,11 +1912,11 @@ pub unsafe extern "C" fn XML_ExternalEntityParserCreate(
     let mut oldDeclElementType: *mut ELEMENT_TYPE = 0 as *mut ELEMENT_TYPE;
     let mut oldUserData: *mut c_void = 0 as *mut c_void;
     let mut oldHandlerArg: *mut c_void = 0 as *mut c_void;
-    let mut oldDefaultExpandInternalEntities: XML_Bool = 0;
+    let mut oldDefaultExpandInternalEntities: XML_Bool = false;
     let mut oldExternalEntityRefHandlerArg: XML_Parser = 0 as *mut XML_ParserStruct;
     let mut oldParamEntityParsing: XML_ParamEntityParsing = XML_PARAM_ENTITY_PARSING_NEVER;
     let mut oldInEntityValue: c_int = 0;
-    let mut oldns_triplets: XML_Bool = 0;
+    let mut oldns_triplets: XML_Bool = false;
```

Comment:
For many of these you can remove the variable type and let type inference do its job.
```suggestion
    let mut oldns_triplets = false;
```